integrationtests/src/test/java/io/kroxylicious/proxy/KrpcFilterIT.java

```diff
@@ -181,7 +195,11 @@ public void shouldOfferPrometheusMetricsScrapeEndpoint() throws Exception {
             Metrics.counter(counter_name).increment();
             HttpRequest request = HttpRequest.newBuilder(URI.create("http://localhost:9193/metrics")).GET().build();
             HttpResponse<String> response = HttpClient.newHttpClient().send(request, ofString());
-            assertTrue(response.body().contains(counter_name + " 1.0"));
+            assertTrue(response.body().contains(counter_name + "{zone=\"euc-1a\",}" + " 1.0"));
+            // from binder
+            assertTrue(response.body().contains("jvm_gc_memory_allocated_bytes_total{zone=\"euc-1a\",}"));
```

Comment:
Personally I think we should have separate test cases for each of the `default labels`, `binder` and `config hook`. 

I'd also prefer 
```suggestion
            org.assertj.core.api.Assertions.assertThat(response.body()).contains("jvm_gc_memory_allocated_bytes_total");
```

Note that deliberately doesn't include the labels so that further changes to the default set don't affect this test.

Reply:
Have hit a bit of a snag on this one.

The micrometer global registry is static and has some configuration you can't clean out between tests. So the tests were non-deterministic, like if you have a separate test for common tags, then depending on test order you see the tags unexpectedly in other tests.

Reply:
Pulled it into it's own test class with a method testing each feature, and made the assertions label agnostic. I also configured surefire and failsafe to use random test ordering while I was at it, to show up problems like that.

Reply:
Keen to go back and do further cleaning in the tests, there's a lot of duplicate creation of kafka brokers and proxies. But I figure touching that conflicts with the coming changes extracting the broker creation?